### PR TITLE
Use another build stage for Travis

### PR DESCRIPTION
**Pull request by kidder**
## Proposed changes

Need to use three stages instead of two to avoid timeouts

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [x] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: .travis/BuildLinux.sh [hunk 1]
```diff
@@ -57,12 +57,22 @@ if [ -z "${RUN_CLANG_TIDY}" ] \
     && [ -z "${RUN_IWYU}" ] \
     && [ -z "${BUILD_DOC}" ]; then
     if [[ ${TRAVIS_BUILD_STAGE_NAME} = "Build libraries" ]]; then
-        time make libs test-libs-stage1 -j2
+        time make libs -j2
+        time make test-libs-domain -j2
```
Comment (nilsdeppe) [inline .travis/BuildLinux.sh:61]: Why not use the `test-libs-stage1` target here?

**Review by nilsdeppe — COMMENTED** [review]
looks good, just one question

File: .travis/BuildLinux.sh (hunk 1 above)
Comment (kidder) [reply .travis/BuildLinux.sh:61]: I thought it would be more interesting to track the timing of smaller groups of test libraries

Comment (nilsdeppe) [reply .travis/BuildLinux.sh:61]: Okay, then can we remove the `test-libs-stage1` target?

**Review by nilsdeppe — APPROVED** [review]
Restarted GCC7 debug

**Comment by kidder** [comment]
@wthrowe please look at

**Review by wthrowe — APPROVED** [review]

**Pull request merged**